quesma/model/select.go

```diff
@@ -11,11 +11,14 @@ type SelectCommand struct {
 	GroupBy     []Expr        // if not empty, we do GROUP BY GroupBy...
 	OrderBy     []OrderByExpr // if not empty, we do ORDER BY OrderBy...
 
-	Limit       int // LIMIT clause, noLimit (0) means no limit
-	SampleLimit int // LIMIT, but before grouping, 0 means no limit
+	LimitBy     []Expr // LIMIT BY clause (empty => maybe LIMIT, but no LIMIT BY)
+	Limit       int    // LIMIT clause, noLimit (0) means no limit
+	SampleLimit int    // LIMIT, but before grouping, 0 means no limit
+
+	Subqueries []SelectCommand
```

Comment:
What kind of query do you want to express with `Subqueries`? Currently, we have `Columns`, `FromClause` and `WhereClause`, all of them are expressions and can be subqueries

Reply:
I do a lot of
```
WITH subquery_1 AS (
  SELECT ...
), subquery_2 AS (
  SELECT ...
)
SELECT ...
FROM table INNER JOIN subquery_1 INNER JOIN subquery_2
```

Reply:
ok, then maybe better name will be `CTE` - `Common Table Expressions` or `WithClause`

Reply:
great, thanks! I'll change to CTE